core/CMakeLists.txt

```diff
@@ -38,6 +38,7 @@ endif()
 
 ginkgo_compile_features(ginkgo)
 
+target_compile_options(ginkgo PRIVATE "-Wpedantic")
```

Comment:
I am not a fan of adding this `-Wpedantic` directly here. I would prefer a varialbe or something similar to be defined in the root CMake file, where it can be set depending on the used compiler and altered in a single place for every part of ginkgo (e.g. if we decide we want to add `-Wconversion` or similar).

Reply:
Yeah, that is better. Do you think it is better to create three separate variables `GINKGO_CORE_WARNING_FLAGS`, `GKO_REFERENCE_WARNING_FLAGS` and similar or one variable `GINKGO_WARNING_FLAGS` would be better ?

Reply:
I don't think we should call them `WARNING_FLAGS`, but instead `COMPILE_FLAGS`, as we might add some non-warning related flags. For now, a single variable could be enough, although I have no idea how they are forwarded in the CUDA case (if they are forwarded to both nvcc and host compiler, or just to one of them)...